### ISSUE TITLE
Updating the stack always puts us back to the MinSize of the ASG

### ISSUE THREAD

**Issue by deppy** [buildkite/elastic-ci-stack-for-aws#260]
The CloudFormation template looks like it was going to allow people to configure the DesiredCapacity (parameter added), but then was hard coded to use the MinSize.  Our MinSize is 0, so any upgrades during the middle of the work day always sets the ASG back to 0 nodes until the CloudWatch alarms slowly have it scale up.  

I'll submit a PR to fix this as it should be configurable by parameter to prevent these cases.

**Comment by gugahoi** [comment]
if you have a desired capacity that is bigger than the minimum size, when will it ever go down to minimum?

**Comment by gugahoi** [comment]
Why would it matter so much? Just let it scale up when workload is needed. Whats the point of a desired capacity in this case other than the first time the ASG starts?

**Comment by deppy** [comment]
Desired capacity is how many instances it should currently have, which is why it matters so much.  If I have `MinSize: 0`, `MaxSize: 20` and the stack currently has `DesiredCapcity: 18` and I update the stack to make any kind of change, the current setting would drop DesiredCapacity back to 0, meaning that I have no agents.  With our current scaling policies, it would take about an hour to go back to the DesiredCapacity of 18.  This means a lot of unhappy developers for an hour (or me manually going in and setting DesiredCapcity on the ASG to 18, when I should be able to do it through the stack).

**Comment by lox** [comment]
I think this is probably being driven by https://github.com/buildkite/elastic-ci-stack-for-aws/blob/master/templates/buildkite-elastic.yml#L455-L467:

```
    UpdatePolicy:
      AutoScalingRollingUpdate:
        MinInstancesInService: 0
        MaxBatchSize: 5
        # On rollback this might have to wait for an agent to finish
        # a job. The agent's init script waits 30 minutes, so we wait
        # a maximum of 40 minutes to be safe.
        PauseTime: PT40M
        WaitOnResourceSignals: true
        # Ignore scaling alarms during update
        SuspendProcesses:
          - AlarmNotification
          - ScheduledActions
```

I need to do some more research on this, I suspect we are missing something here.

**Comment by lox** [comment]
This was closed by #261, with some follow-up work pending in #270.
